Fix L<> entries without sections

richsalz opened the pull request:
Add sections (almost always "(3)" to L<> references that were missing
them.

Also remove two references to non-existant manpages that have never
existed, and with the 3.0 structure, are unlikely to do so.

Replaces #10097

levitte commented on doc/man3/OSSL_CMP_CTX_new.pod:
```diff
@@ -532,7 +532,8 @@ CertRepMessage or Revocation Response or error message, or NULL if unset.
 of the last received CertRepMessage or Revocation Response or error message.
-This is a bit field and the flags for it are specified in L<cmp.h>.
+This is a bit field and the flags for it are specified in the header file
+C<cmp.h>.
```
That should be `F<< <openssl/cmp.h> >>`

richsalz replied:
Done.  It is nice to see that some practices have improved since this PR was first created.

levitte commented on util/find-doc-nits:
```diff
@@ -447,6 +447,24 @@ sub check {
+    # Make sure every link has a section.
+    while ( $contents =~ /$markup_re/msg ) {
+        my $target = $1;
+        next unless $target =~ /L</;            # Skip if not L<...>
```
I would make that re `/^L</`

richsalz replied:
done.

levitte commented on util/find-doc-nits:
```diff
@@ -447,6 +447,24 @@ sub check {
+        next if $target =~ m@L</@;              # Links within the page
+        next if $target =~ /^L<https?:/;        # URL link
+        next if $target =~ m@\([1357]\)>$@;     # Has a section
+        next if $target =~ m@\([1357]\)/[^>].*>$@; # Has a section/anchor
```
`[^>]` is superfluous at this point

richsalz replied:
yeah, when "*" become ".*" I missed that.  fixed.

richsalz commented:
Fixup commit pushed.  I also added comments to lines 453ff.

levitte commented on doc/man3/OSSL_CMP_CTX_new.pod:
```diff
@@ -532,7 +532,8 @@ CertRepMessage or Revocation Response or error message, or NULL if unset.
 of the last received CertRepMessage or Revocation Response or error message.
-This is a bit field and the flags for it are specified in L<cmp.h>.
+This is a bit field and the flags for it are specified in the header file
+F<< <cmp.h> >>.
```
Sorry, this is incorrect. The official header file is `<openssl/cmp.h>`, not `<cmp.h>`

richsalz replied:
I got lost in the set of angle brackets :)  I notice that the next line has some text that should be marked up somehow, but for another PR.

richsalz commented:
Rebased and fixed another nit in the just-added doc/man7/EVP_KDF-KRB5KDF.pod:
```
; g diff master doc/man7/EVP_KDF-KRB5KDF.pod 
diff --git a/doc/man7/EVP_KDF-KRB5KDF.pod b/doc/man7/EVP_KDF-KRB5KDF.pod
index 08f50d1..6291a11 100644
--- a/doc/man7/EVP_KDF-KRB5KDF.pod
+++ b/doc/man7/EVP_KDF-KRB5KDF.pod
@@ -51,7 +51,7 @@ parameter to the L<EVP_KDF_derive(3)> function, and MUST match the key
 length for the chosen cipher or an error is returned. Moreover the
 constant's length must not exceed the block size of the cipher.
 Since the KRB5KDF output length depends on the chosen cipher, calling
-L<EVP_KDF_size()> to obtain the requisite length returns the correct length
+L<EVP_KDF_size(3)> to obtain the requisite length returns the correct length
 only after the cipher is set. Prior to that B<EVP_MAX_KEY_LENGTH> is returned.
 The caller must allocate a buffer of the correct length for the chosen
 cipher, and pass that buffer to the L<EVP_KDF_derive(3)> function along
@@ -94,7 +94,7 @@ RFC 3961
 
 =head1 SEE ALSO
 
-L<EVP_KDF>,
+L<EVP_KDF(3)>,
 L<EVP_KDF_CTX_new_id(3)>,
 L<EVP_KDF_CTX_free(3)>,
 L<EVP_KDF_ctrl(3)>,
; 
```

t8m commented:
> Rebased and fixed another nit in the just-added doc/man7/EVP_KDF-KRB5KDF.pod:

Did you push it?

richsalz commented:
did now. :)

levitte commented:
Did not...

richsalz commented:
I didn't come here for an argument :)  (Yes you did...)  Sorry, pushed from the wrong spot.  *NOW* it's fixed.

levitte commented:
So, er, is this in an approvable state now, or do you plan on more small fixes? If the latter, I can wait...

richsalz commented:
I don't have plans to make any changes, unless someone in this PR finds a bug with it.

mspncp commented:
Re-review requested, because branch was force-pushed after approval.

levitte commented:
Did the force-push make any relevant changes to this PR?  @richsalz has apparently made it a hobby to randomly rebase-and-push...

richsalz commented:
Hobby was joking around :)  I like to keep my branches current with master, rather than wait for conflicts.  I am trying to move to a "wait for conflict to appear" indicator.

This force-push was rebase only, no changes.

levitte commented:
I have told you already over in the other PR, how that can be disruptive. Here, you got to see that live. We have also started to enforce the 24h grace period, so every time you make an unnecessary force-push, you risk delaying the merge of this PR by 24 more hours. I'm sure that's not an effect you desire. 

So, if the force-pushes since my last approval didn't change anything in this PR, my approval as such stands. @t8m's approval is older, he will have to say if it still stands.

richsalz commented:
I had rebased locally, and then because Matt wanted me to add a Fixes line, I had to push.
It will take awhile to unlearn my fingers and not rebase all branches all the time.

mspncp commented:
> I had rebased locally, and then because Matt wanted me to add a Fixes line, I had to push.
> It will take awhile to unlearn my fingers and not rebase all branches all the time.

FWIW: You can do an interactive rebase without moving the merge-base by explicitly specifying the hash of the merge-base instead of 'master' (`git rebase -i <merge-base-id>` ). In that case the [force-pushed](https://github.com/openssl/openssl/compare/85da3a0837ac814f577cfdc603faad0facd610b0..ce473062580e171aea6018fa4c964e94a0315836) view which is automatically created by GitHub will show only your changes and not those of master, too.

t8m commented:
My approval still stands as well.

t8m commented:
Ah, so I see @mspncp cleared the Richard's review status. @levitte could you please formally re-approve?

levitte commented:
Uhmmm, no-one has cleared my approval. It's still there, and I subsequently asserted that it still stands. Do I have to re-approve? From a procedure point of view, being ok with my previous approval would make this PR mergable *now*, while a new approval delays that by another 24 hours.

t8m commented:
@levitte See the yellow dot at the reviewer list.
'mspncp requested a review from levitte 2 days ago'

levitte commented:
So my assertion that my approval stands, which I made after that request (https://github.com/openssl/openssl/pull/10240#issuecomment-547566182), isn't appropriate enough?

t8m commented:
I do not know - will addrev handle it? If so, I do not have a problem with it.

levitte commented:
Addrev currently doesn't look at the delay thing, and I don't see that it should.

richsalz commented:
> Addrev currently doesn't look at the delay thing, and I don't see that it should.

I totally agree.  For example, during the final stages of release prep, it seems doubtful that you would want to enforce the 24hour delay.

t8m commented:
The conflict was trivial (just a file rename). Merged to master as 6e4618a0d7.
